frontend/packages/console-dynamic-plugin-sdk/src/api/flag.ts

```diff
@@ -0,0 +1,34 @@
+import { Map as ImmutableMap } from 'immutable';
+// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+// @ts-ignore: FIXME missing exports due to out-of-sync @types/react-redux version
+import { useSelector } from 'react-redux';
+import { featureReducerName } from '../constants';
+
+export const useFlag: UseFlag = (flag) =>
+  useSelector<RootState, boolean>(({ FLAGS }) => FLAGS.get(flag));
+
+type UseFlag = (flag: string) => boolean;
+
+type Request<R> = {
```

Comment:
Can we move this type (and others below) into `packages/console-dynamic-plugin-sdk/src/extensions/console-types.ts` ?